File: modulo-2/aula-1/ecommerce/produtos.py
# Sistema de Gerenciamento de Produtos com bugs intencionais para exercícios

class GerenciadorProdutos:
    def __init__(self):
        self.produtos = {}

    # Bug 1: Não valida preço negativo
    def cadastrar_produto(self, id, nome, preco, estoque):
        self.produtos[id] = {
            'nome': nome,
            'preco': preco,
            'estoque': estoque
        }
        return True

    # Bug 2: Não verifica se produto existe
    def buscar_produto(self, id):
        return self.produtos.get(id)

    # Bug 3: Não atualiza estoque corretamente
    def atualizar_estoque(self, id, quantidade):
        produto = self.produtos.get(id)
        produto['estoque'] = quantidade
        return True

    # Bug 4: Lista produtos sem estoque
    def listar_produtos(self):
        return list(self.produtos.values())

    # Bug 5: Não valida dados de entrada
    def atualizar_produto(self, id, dados):
        produto = self.produtos.get(id)
        produto.update(dados)
        return True 

File: modulo-2/aula-1/ecommerce/produtos.js
// Sistema de Gerenciamento de Produtos com bugs intencionais para exercícios

class GerenciadorProdutos {
    constructor() {
        this.produtos = new Map();
    }

    // Bug 1: Não valida preço negativo
    cadastrarProduto(id, nome, preco, estoque) {
        this.produtos.set(id, {
            nome: nome,
            preco: preco,
            estoque: estoque
        });
        return true;
    }

    // Bug 2: Não verifica se produto existe
    buscarProduto(id) {
        return this.produtos.get(id);
    }

    // Bug 3: Não atualiza estoque corretamente
    atualizarEstoque(id, quantidade) {
        const produto = this.produtos.get(id);
        produto.estoque = quantidade;
        return true;
    }

    // Bug 4: Lista produtos sem estoque
    listarProdutos() {
        return Array.from(this.produtos.values());
    }

    // Bug 5: Não valida dados de entrada
    atualizarProduto(id, dados) {
        const produto = this.produtos.get(id);
        Object.assign(produto, dados);
        return true;
    }
}

module.exports = GerenciadorProdutos; 